### PR TITLE
Detect relevant IbcRelay actions

### PR DESCRIPTION
**Pull request by grod220**
At the moment, after a user makes a Ics20Transfer, Prax is not indexing the transaction for the IbcRelay action. This can cause confusion for users wondering the status of transaction.

This PR detects the relay action and, if the user is the recipient, adds it to the tx db.

### CODE REVIEW

File: packages/protobuf/src/registry.ts [hunk 1]
```diff
@@ -41,6 +42,9 @@ export const typeRegistry: IMessageTypeRegistry = createRegistry(
   // gen/ibc/core/channel/v1/tx_pb
   MsgRecvPacket,
 
+  // gen/ibc/core/client/v1/tx_pb
+  MsgUpdateClient,
+
```
Comment (grod220) [inline packages/protobuf/src/registry.ts:47]: Required to add the the type registry as an IbcRelay transaction sent from the extension to the frontend needs to be able to serialize the `Any` inside of it & a `MsgUpdateClient` is accompanied with a `MsgRecvPacket`

**Comment by grod220** [comment]
Some type registry updates are needed for this PR. Will re-request review when ready.

File: packages/ui/components/ui/tx/actions-views/ibc-relay.tsx
Comment (grod220) [inline packages/ui/components/ui/tx/actions-views/ibc-relay.tsx:1]: Screenshot 2024-09-24 at 11 19 12 AM

File: apps/minifront/src/components/tx-details/tx-viewer.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export const TxViewer = ({ txInfo }: { txInfo?: TransactionInfo }) => {
 
   // use React-Query to invoke custom hooks that call async translators.
   const { data: receiverView } = useQuery(
-    ['receiverView', txInfo, option],
+    ['receiverView', txInfo?.toJson({ typeRegistry }), option],
```
Comment (grod220) [inline apps/minifront/src/components/tx-details/tx-viewer.tsx:51]: Now that we are saving relay transactions, there is a change of an `Any` field needing to be serialized. For that reason, it needs access to the registry to know how to serialize/deserialize. Hence, why you see `typeRegistry` being added to a number of from/toJson() methods.

File: packages/protobuf/src/registry.ts [hunk 3]
```diff
@@ -38,9 +36,6 @@ export const typeRegistry: IMessageTypeRegistry = createRegistry(
   ClientState,
   Header,
 
-  // gen/ibc/core/channel/v1/tx_pb
-  MsgRecvPacket,
-
```
Comment (grod220) [inline packages/protobuf/src/registry.ts:43]: Not necessary to list specifically in the registry as it exists in `IbcChannelMsgService` already

File: packages/ui/components/ui/tx/actions-views/ibc-relay.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+const parsePacket = ({ packet }: MsgRecvPacket): FungibleTokenPacketData | undefined => {
+  if (!packet?.data) {
+    return undefined;
+  }
+
+  try {
+    const dataString = new TextDecoder().decode(packet.data);
+    return FungibleTokenPacketData.fromJsonString(dataString);
+  } catch (e) {
+    return undefined;
```
Comment (vacekj) [inline packages/ui/components/ui/tx/actions-views/ibc-relay.tsx:24]: Suggestion: should we error-log here, or display something to the user even?

**Review by vacekj — APPROVED** [review]
lgtm

File: packages/ui/components/ui/tx/actions-views/ibc-relay.tsx (hunk 4 above)
Comment (grod220) [reply packages/ui/components/ui/tx/actions-views/ibc-relay.tsx:24]: Hmm, perhaps we do. I'll handle the case where the packet cannot be parsed and display `Unknown packet data` in the row.

**Pull request merged**